src/pothole_detection/Potholes/generate.py

```diff
@@ -120,28 +125,32 @@ def draw_ellipse(input_path, max_potholes):
             background = background.filter(
                 ImageFilter.GaussianBlur(random.randint(50, 125)/100))
             # Scale down image to get a choppier ellipse
-            scale = random.randint(30, 60) / 100
-            mask = Image.new("L", ((int)(image.width * scale),
-                             (int)(image.height * scale)), 0)
+            # scale = random.randint(30, 60) / 100
+            # mask = Image.new("L", ((int)(image.width * scale),
+            #                  (int)(image.height * scale)), 0)
+            # ImageDraw.Draw(mask).ellipse(
+            #     ((int)(x_start * scale), (int)(y_start * scale),
+            #      (int)(x_end * scale), (int)(y_end * scale)), fill=255)
+            # mask = mask.resize((image.width, image.height))
+
+            # mask = mask.filter(ImageFilter.GaussianBlur(random.randint(1, 4)))
+            mask = Image.new("L", image.size, 0)
             ImageDraw.Draw(mask).ellipse(
-                ((int)(x_start * scale), (int)(y_start * scale),
-                 (int)(x_end * scale), (int)(y_end * scale)), fill=255)
-            mask = mask.resize((image.width, image.height))
-
-            mask = mask.filter(ImageFilter.GaussianBlur(random.randint(1, 4)))
+                (x_start, y_start, x_end, y_end), fill=255)
             image = Image.composite(background, image, mask)
             potholes.append([loc_x, loc_y, x_rad * 2, y_rad * 2, obj_class])
             count += 1
     if len(potholes) == 0:
         return
-    image.save(os.path.join(DATA_OUT, Path(input_path).name))
+    image.save(os.path.join(DATA_OUT, os.path.splitext(
+        Path(input_path).name)[0] + 'D.png'))
     height, width, rgb = np.array(image).shape
     # Normalize potholes locations
     potholes = [[(float)(x[0])/width, (float)(x[1])/height,
                  (float)(x[2])/width, (float)(x[3])/height, x[4]
                  ] for x in potholes]
     # Save the label
```

Comment:
Just a general comment, instead of commenting _what_ you're doing, maybe split this into a function with a descriptive name: `save_label()`. Helps code clarity + shortens the function!

I'm guilty of this too 😇